extensions/amp-story/1.0/amp-story.js

```diff
@@ -1455,6 +1460,39 @@ export class AmpStory extends AMP.BaseElement {
     }
   }
 
+  /**
+   * @private
+   */
+  openMask_() {
+    const actions = Services.actionServiceForDoc(this.getAmpDoc());
+    if (!this.maskElement_) {
+      const mask = this.win.document.createElement('div');
+      mask.classList.add('i-amphtml-amp-story-sidebar-mask');
+      mask.addEventListener('click', () => {
+      actions.execute(this.sidebar_, 'close', /* args */ null,
+        /* source */ null, /* caller */ null, /* event */ null,
+            ActionTrust.HIGH);
+      });
+      // The mask div should be added right after the sidebar.
+      const referenceNode = this.win.document.getElementsByTagName('amp-sidebar')[0];
+      referenceNode.parentNode.insertBefore(mask, referenceNode.nextSibling);
```

Comment:
Given the `z-index` that you set, you should be able to append the mask at any position as long as it's a direct child of `amp-story` (which you can refer to as `this.element` since you're in the amp-story component).
I believe you could do: `this.element.appendChild(maskEl)`